CRM/Upgrade/Incremental/php/SixOne.php

```diff
@@ -29,6 +29,29 @@ class CRM_Upgrade_Incremental_php_SixOne extends CRM_Upgrade_Incremental_Base {
    */
   public function upgrade_6_1_alpha1($rev): void {
     $this->addTask(ts('Upgrade DB to %1: SQL', [1 => $rev]), 'runSql', $rev);
+    $this->addExtensionTask('Enable Riverlea extension', ['riverlea']);
+    $this->addTask('Freeze "default" theme to Greenwich', 'freezeDefaultThemeToGreenwich');
+  }
+
+  /**
+   * 'default' is being removed as a valid theme choice.
+   *
+   * It has always meant Greenwich, so sites previously on
+   * "default" change this to an explicit value Greenwich.
+   *
+   * @return bool
+   */
+  public static function freezeDefaultThemeToGreenwich() {
+    $themeSettings = ['theme_backend', 'theme_frontend'];
+
+    foreach ($themeSettings as $key) {
+      $value = \Civi::settings()->get($key);
+      if ($value === 'default') {
+        \Civi::settings()->set($key, 'greenwich');
```

Comment:
Should this be SQLised?

Reply:
Blerg. The current code in `CRM/Upgrade/Incremental/` is divided, and their are theoretical risks either way.

TBH, the `Civi::settings()` subsystem is, eh, sui generis... It's called during bootstrap; values are always prefetched into PHP-memory; the code is lower-level; the schema doesn't change much; and when it has changed, we wound up with extra guards in `SettingsBag.php`. (If you don't teach `SettingsBag` to work during upgrade-mode, then you're likely to crash before the upgrade runs.) _If you can get to the point of running these steps_, then the `settings` subsystem should be working.

I think it's OK as-is.

Reply:
Ok great, thanks @totten